### PR TITLE
fix(multi_object_tracker): fix knownConditionTrueFalse

### PR DESCRIPTION
**Pull request by veqcc**
## Description

This is a fix based on cppcheck `knownConditionTrueFalse` warings

```
perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp:70:7: style: Condition 'sparse_cost' is always true [knownConditionTrueFalse]
  if (sparse_cost) {
      ^
perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp:51:22: note: Assignment 'sparse_cost=true', assigned value is 1
  bool sparse_cost = true;
                     ^
perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp:61:24: note: Assuming condition is false
  if (cost.size() == 0 || cost.at(0).size() == 0) {
                       ^
perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp:70:7: note: Condition 'sparse_cost' is always true
  if (sparse_cost) {
      ^
```

@technolojin 
If this is an acceptable change, then I will make the same changes on `object_merger`, `radar_object_tracker` and `tracking_object_merger` packages in this PR.

## Related links

**Parent Issue:**

- Link

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Review by technolojin — CHANGES_REQUESTED** [review]
@veqcc I think this is not the case. 
`sparse_cost` may need to be defined outside the function instead.

**Comment by veqcc** [comment]
@technolojin 
Thank you for your quick response!

> @veqcc I think this is not the case.
sparse_cost may need to be defined outside the function instead.

Understood.
Do you have any idea for avoiding cpocheck warnings, or is there any plan to refactor these packages so that `sparse_cost` is not hard coded any more?

**Comment by veqcc** [comment]
@technolojin @YoshiRi @yukkysaito 
Hi

File: perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp [hunk 1]
```diff
@@ -47,10 +47,6 @@ void SSP::maximizeLinearAssignment(
   const std::vector<std::vector<double>> & cost, std::unordered_map<int, int> * direct_assignment,
   std::unordered_map<int, int> * reverse_assignment)
 {
-  // NOTE: Need to set as default arguments
```
Comment (YoshiRi) [inline perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp:50]: With this statement, I think it's better to define sparse_cost in the arguments and give default true value.

File: perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp [hunk 2]
```diff
@@ -47,10 +47,6 @@ void SSP::maximizeLinearAssignment(
   const std::vector<std::vector<double>> & cost, std::unordered_map<int, int> * direct_assignment,
   std::unordered_map<int, int> * reverse_assignment)
```
Comment (YoshiRi) [inline perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp:48]:
```suggestion
  std::unordered_map<int, int> * reverse_assignment, const bool sparse_cost = true)
```

With the Notation in the original code, how about this? ( Need .hpp changes )

**Review by YoshiRi — CHANGES_REQUESTED** [review]
How about put the variable into arguments with default value?

File: perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp (hunk 1 above)
Comment (veqcc) [reply perception/multi_object_tracker/src/data_association/successive_shortest_path/successive_shortest_path.cpp:50]: Thank you for your advice!! I will accepr your suggestion.

**Comment by veqcc** [comment]
I will close this PR and create a new PR in which I will solve all the warnings around `sparse_cost` in object_merger, radar_object_tracker and tracking_object_merger with the same way here
> With this statement, I think it's better to define sparse_cost in the arguments and give default true value.

**Pull request closed without merging**